dask/highlevelgraph.py

```diff
@@ -480,7 +546,7 @@ def __init__(
         self,
         layers: Mapping[str, Mapping],
         dependencies: Mapping[str, AbstractSet],
-        key_dependencies: Dict[Hashable, AbstractSet] = None,
+        key_dependencies: Optional[Dict[Hashable, AbstractSet]] = None,
```

Comment:
nope, it was correct before (aligned to typeshed guidelines)

Reply:
Not that I have any strong opinion about this but doesn't typeshed specifically state that `Optional` should be used: https://github.com/python/typeshed/blob/master/CONTRIBUTING.md ?

Reply:
My bad. I strongly prefer the implicit version, as IMHO the explicit variant is completely redundant and needlessly verbose.

Reply:
I only changed it because `mypy --no-implicit-optional` will complain if `Optional` isn't used. 
But I agree, let's go with the implicit version.

Reply:
Hopefully [PEP-645](https://www.python.org/dev/peps/pep-0645/) will make this a bit more ergonomic.